src/graphs/Dijkstra.re

```diff
@@ -0,0 +1,151 @@
+type node = {
+    id: string,
+    neighbours: list(string),
+};
+
+type directedGraph = list(node);
+
+type edge = {
+    tail: string,
+    head: string,
+    weight: int,
+};
+
+type resultType = {
+    dist: Hashtbl.t(string, int),
+    prev: Hashtbl.t(string, option(string))
+};
+
+exception Not_found(string);
+
+let infinity = Pervasives.max_int;
+
+let parseAdjList = adj_list => {
+    let adj_tbl = Hashtbl.create(List.length(adj_list));
+    let insert = ({id, neighbours}) => {
+        Hashtbl.add(adj_tbl, id, neighbours);
+    };
+
+    List.iter(insert, adj_list);
+    let validateNeighbours = node => {
+        List.iter(neighbour => {
+            if (!Hashtbl.mem(adj_tbl, neighbour)) {
+                raise(Not_found(neighbour));
+            }
+        }, node.neighbours);
+    };
+
+    List.iter(validateNeighbours, adj_list);
+    adj_tbl;
+};
+
+let parseWeights = (weight_list, adj_list) => {
+    let weight_tbl = Hashtbl.create(List.length(adj_list));
+    let initializeWeights = ({id, neighbours}) => {
+        let adj_edge_weights = Hashtbl.create(List.length(neighbours));
+        Hashtbl.add(weight_tbl, id, adj_edge_weights);
+    };
+
+    List.iter(initializeWeights, adj_list);
+    let insertWeight = ({tail, head, weight}) => {
+        let adj_edge_weights = Hashtbl.find(weight_tbl, tail);
+        Hashtbl.add(adj_edge_weights, head, weight);
+    };
+
+    List.iter(insertWeight, weight_list);
+    weight_tbl;
+};
+
+let initializeDist = (adj_list, source_id) => {
+    let dist = Hashtbl.create(List.length(adj_list));
+    let initialize = ({id, _}) => {
+        Hashtbl.add(dist, id, infinity);    
+    };
+
+    List.iter(initialize, adj_list);
+    Hashtbl.replace(dist, source_id, 0);
+    dist;
+};
+
+let initializePrev = adj_list => {
+    let prev = Hashtbl.create(List.length(adj_list));
+    let initialize = ({id, _}) => {
+        Hashtbl.add(prev, id, None);
+    };
+
+    List.iter(initialize, adj_list);
+    prev;
+};
+
+let getEdgeWeight = (weight_tbl, tail, head) => {
+    let adj_edge_weights = Hashtbl.find(weight_tbl, tail);
+    Hashtbl.find(adj_edge_weights, head);
+};
+
+let add = (a, b) => {
+    if (a == infinity || b == infinity) {
```

Comment:
I don't think we should use infinity. `Pervasive.infinity` is for floating points but here we are dealing with `int`. The distance hash_tbl can map to `option(int)`

Reply:
I used `let infinity = Pervasives.max_int;` not `let infinity = Pervasives.infinity`. 
So should I still change this?

Reply:
debatable but i think it's fine.